pkg/api/handlers/compat/containers.go

```diff
@@ -44,11 +45,24 @@ func RemoveContainer(w http.ResponseWriter, r *http.Request) {
 	runtime := r.Context().Value("runtime").(*libpod.Runtime)
 	name := utils.GetName(r)
 	con, err := runtime.LookupContainer(name)
-	if err != nil {
-		utils.ContainerNotFound(w, name, err)
-		return
-	}
+	if err != nil && errors.Cause(err) == define.ErrNoSuchCtr {
+		// Failed to get container. If force is specified, get the container's ID
+		// and evict it
+		if !query.Force {
+			utils.ContainerNotFound(w, name, err)
+			return
+		}
 
+		if _, err := runtime.EvictContainer(r.Context(), name, query.Vols); err != nil {
```

Comment:
I'm still fairly certain the `err == nil` case isn't handled properly - we should be returning immediately with a 204 in that case.